components/wifi/COMPONENT_QCA400X_WIFI/QCA_WiFiInterface.cpp

```diff
@@ -0,0 +1,636 @@
+/* QCA4002/4004 implementation of NetworkInterfaceAPI
+ * Copyright (c) 2019 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "QCA_WiFiInterface.h"
+#include "events/EventQueue.h"
+#include "events/mbed_shared_queues.h"
+#include "features/netsocket/nsapi_types.h"
+#include "mbed_trace.h"
+#include "platform/Callback.h"
+#include "platform/mbed_critical.h"
+#include "platform/mbed_debug.h"
+#include "platform/mbed_wait_api.h"
+#include <stdio.h>
+#define TRACE_GROUP  "QCAI" // ESP8266 Interface
+
+using namespace mbed;
+
+QCA_WiFiInterface::QCA_WiFiInterface():
```

Comment:
Thanks for adding the `400X` part. Do we need the `Wifi` part for any reason. I also wonder about the underscore. I read the [Coding style guidelines](https://os.mbed.com/docs/mbed-os/v5.13/contributing/style.html) and all it says is that every word should start with a capital letter. Underscored are not explicitly forbidden, but I don't think this underscore can be justified in here?
I am personally a fan of `QCA400XInterface` - the `Wifi` part just seems redundant (unless there is a module with the same name, that is not a Wifi module?)
@0xc0170 , @VeijoPesonen,  any toughts on this?

Reply:
We do have [OdinWiFiInterface](https://github.com/ARMmbed/mbed-os/blob/8ef742a49c1682f9ef3ba50148b871e38c3866cc/targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F439xI/TARGET_MODULE_UBLOX_ODIN_W2/sdk/ublox-odin-w2-drivers/OdinWiFiInterface.h)

Reply:
@VeijoPesonen , `Odin` does not clearly indicate a wifi chip, but a board family, so I imagine it might have more drivers and related `Odin*Interface` classes. QCA400x is [a WiFi-only chip](https://developer.qualcomm.com/hardware/qca4002-4), so it will probably have no other `Interface` than a wifi one :).
I think `Wifi` is redundant, but if noone else sees this is as a problem, I don't want to hold back the review. Let's just clarify the mbed_lib.json name, that I mentioned in the other comment.

Reply:
I will change to drop the Wifi from the name